### PR TITLE
fix(normalization): Skip validation on renormalization

### PR DESCRIPTION
**Pull request by iker-barriocanal**
Previous normalization PRs run validation from librelay calls. However, some `sentry` tests run full normalization in librelay on partial (i.e. invalid) payloads, and running validation makes plenty of tests fail (see https://github.com/getsentry/sentry/pull/66134).

Similarly, many `sentry` tests expect span normalization to never happen. After spending significant time "fixing"* some tests, it'd require me even more time to "fix" the remaining ones as it requires a deeper understanding of other features/codebases. Thus, I'm reintroducing `normalize_spans` to disable span normalization (mostly exclusive time computation) from librelay calls and unblock the librelay release.

*Note: "fixing" a test means using data relay outputs from normalization. This doesn't align with other assumptions made in the tests, nor modifications to the event payload in the rest of the event processing pipeline.

### Future work
- Consider span normalization a regular step in event normalization behind the `is_renormalize` flag, removing the `normalize_spans` flag.

### CODE REVIEW

File: relay-event-normalization/src/validation.rs
Comment (olksdr) [inline relay-event-normalization/src/validation.rs]: Have you considered to name this differently to make sure it's clear what the option means without reading the docs? 

To be honest even docs do not describe now what the option should be set to disable validation.

Maybe? 
```suggestion
    pub is_validated: bool
```

Comment (iker-barriocanal) [reply relay-event-normalization/src/validation.rs]: I chose `is_renormalize` because it's a known name and concept we use in normalization and librelay. Is `is_validated` more clear? I can update the name and expand docs if it helps.

**Review by jjbayer — APPROVED** [review]
Approving to unblock. It would be nice to get rid of the config flag eventually.

**Review by Dav1dde — APPROVED** [review]
I don't fully understand the change and it seems to me we're working around bad tests in Sentry, but we are definitely blocked on this.

Have you considered exposing the flag from librelay via Keyword arg, so we have at least a chance of fixing the Sentry tests on a step by step basis?

File: relay-event-normalization/src/validation.rs
Comment (iker-barriocanal) [reply relay-event-normalization/src/validation.rs]: Should be addressed in [bd72909](https://github.com/getsentry/relay/pull/3214/commits/bd729097df816356731b6c53541eea860106119d), let me know if it's still unclear or there's a better wording for it.

**Comment by iker-barriocanal** [comment]
@Dav1dde great idea! Exposed the flag in [380895a](https://github.com/getsentry/relay/pull/3214/commits/380895abe8f8159d1dafc44dfe4046a63da21830). It can be enabled by initing the `StoreNormalizer` with `normalize_spans=True, is_renormalize=False`.

**Pull request merged**
